File: 백준/Bronze/2798. 블랙잭/블랙잭.py
a, b = map(int, input().split())

arr = []

arr = list(map(int, input().split()))

result = 0

for i in range(a):
  for j in range(i + 1, a):
    for k in range(j + 1, a):
      temp = arr[i] + arr[j] + arr[k]
      if temp <= b:
        result = max(result, temp)

print(result)

File: 백준/Bronze/2798. 블랙잭/블랙잭.c
#include <stdio.h>

int getMax(int a, int b){
   if(a > b)   return a;
   else   return b;
}

int main(void) { 
   int a, b;
   scanf("%d %d", &a, &b);
   
   int arr[a];
   int result = 0;
   for (int i = 0; i < a; i++) {
      scanf("%d", &arr[i]);
   }

   for (int i = 0; i < a; i++) {
      for(int j = i + 1; j < a; j++){
         for(int k = j + 1; k < a; k++){
            if(arr[i] + arr[j] + arr[k] <= b)
               result = getMax(result, arr[i] + arr[j] + arr[k]);
         }
      }
   }

   printf("%d", result);
   
   return 0;
}